custom-targets/vertex-ai/advanced_quickstart/ADVANCED_QUICKSTART.md

```diff
@@ -0,0 +1,369 @@
+docker image to run the model in.
+For this quickstart, we will be using a pre-trained model from Vertex AI's cloud samples bucket.
+For the docker image, we use a Cloud AI optimized version of tensorflow: [tf_opt](https://cloud.google.com/vertex-ai/docs/predictions/optimized-tensorflow-runtime).
+
+   ```shell
+   gcloud ai models upload \
+       --artifact-uri gs://cloud-samples-data/vertex-ai/model-deployment/models/boston/model \
+       --display-name=test-model \
+       --container-image-uri=us-docker.pkg.dev/vertex-ai-restricted/prediction/tf_opt-cpu.nightly:latest \
+       --project=$PROJECT_ID \
+       --region=$REGION \
+       --model-id=test_model
+   ```
+
+If this is your first time using Vertex AI in this project, this operation will
+take 10 minutes or so.
+
+## 5. Create Vertex AI Endpoints
+
+Create the Vertex AI endpoints using the following commands:
+
+   ```shell
+   export DEV_ENDPOINT_ID="dev"
+   gcloud ai endpoints create --display-name $DEV_ENDPOINT_ID --endpoint-id $DEV_ENDPOINT_ID --region $REGION --project $PROJECT_ID
+   ```
+
+   ```shell
+   export PROD_ENDPOINT_ID="prod"
+   gcloud ai endpoints create --display-name $PROD_ENDPOINT_ID --endpoint-id $PROD_ENDPOINT_ID --region $REGION --project $PROJECT_ID
+   ```
+
+If this is your first time using Vertex AI in this project, this operation will
+take 5 minutes or so.
+
+The endpoint ID will be used to refer to the endpoint, rather than the display name.
+
+## 6. Create delivery pipeline, targets, and skaffold
+
+Within the `quickstart` directory, run this command to make a temporary copy of `clouddeploy.yaml` and
+`configuration/skaffold.yaml`, and to replace placeholders in the copies with actual values
+
+```shell
+export TMPDIR=$(mktemp -d)
+./replace_variables.sh -p $PROJECT_ID -r $REGION -d $DEV_ENDPOINT_ID -e $PROD_ENDPOINT_ID -t $TMPDIR
+```
+
+The command does the following:
+1. Creates temporary directory $TMPDIR and copies `clouddeploy.yaml` and `configuration` into it.
+2. Replaces the placeholders in `$TMPDIR/clouddeploy.yaml`
+3. Obtains the URL of the latest version of the custom image, built in step 6, and sets it in `$TMPDIR/configuration/skaffold.yaml`
+
+
+Lastly, apply the Cloud Deploy configuration defined in `clouddeploy.yaml`:
+
+```shell
+gcloud deploy apply --file=$TMPDIR/clouddeploy.yaml --project=$PROJECT_ID --region=$REGION
+```
+
+## 7. Create a release and rollout
+
+Create a Cloud Deploy release for the configuration defined in the `configuration` directory. This automatically
+creates a rollout that deploys the first model version to the development target.
+
+```shell
+gcloud deploy releases create release-001 \
+    --delivery-pipeline=vertex-ai-cloud-deploy-pipeline \
+    --project=$PROJECT_ID \
+    --region=$REGION \
+    --source=$TMPDIR/configuration \
+    --deploy-parameters="customTarget/vertexAIModel=projects/$PROJECT_ID/locations/$REGION/models/test_model@1"
+```
+
+### Explanation of command line flags
```

Comment:
nit: hyphenate "command-line" here

Reply:
And every case where "command line" precedes a noun. So: "Invoke a release from the command line." But:  "command-line flag."